src/duckdb.rs

```diff
@@ -602,3 +612,27 @@ fn create_table_function_view_name(table_reference: &TableReference) -> TableRef
     .join(".");
     TableReference::from(&tbl_ref_view)
 }
+
+async fn apply_memory_limit(
+    pool: &Arc<DynDuckDbConnectionPool>,
+    memory_limit: &str,
+) -> DataFusionResult<()> {
+    tracing::debug!("Setting DuckDB memory limit to {memory_limit}");
+
+    if let Err(err) = byte_unit::Byte::parse_str(memory_limit, true) {
```

Comment:
Ensure the parameter is valid before executing it. This prevents arbitrary or malicious SQL from being provided as the memory limit parameter.